lib/system/debugger.nim

```diff
@@ -261,7 +261,7 @@ proc genericHash(dest: pointer, mt: PNimType): int =
 proc dbgRegisterWatchpoint(address: pointer, name: cstring,
                            typ: PNimType) {.compilerproc.} =
   let L = watchPointsLen
-  for i in 0.. <L:
+  for i in 0 .. pred(L):
```

Comment:
did you try to use ..< here?

Reply:
I didn't try, but I don't think it will work because `pred` is defined earlier in system.nim (at around line 600), while `..<` comes very late at around line 3400. The includes for `debugger` and `repr` come at around line 3200 so they either have to use `pred` or we need to rearrange things.

Reply:
Ah, sorry then